src/server/main_service.cc

```diff
@@ -2063,11 +2063,21 @@ GlobalState Service::GetGlobalState() const {
   return global_state_;
 }
 
-void Service::ConfigureHttpHandlers(util::HttpListenerBase* base) {
+void Service::ConfigureHttpHandlers(util::HttpListenerBase* base, bool is_admin) {
   // We set the password for the HTTP service unless it is only enabled on the
   // admin port and the admin port is password-less.
-  if (GetFlag(FLAGS_primary_port_http_enabled) || !GetFlag(FLAGS_admin_nopass)) {
-    base->SetPassword(GetPassword());
+  if (is_admin && GetFlag(FLAGS_admin_nopass)) {
+    base->SetAuthFunctor([](std::string_view path, std::string_view username,
+                            std::string_view password) { return username == "default"; });
+  } else if (GetFlag(FLAGS_primary_port_http_enabled) || !GetFlag(FLAGS_admin_nopass)) {
+    std::string pass = GetPassword();
+    base->SetAuthFunctor([pass = std::move(pass)](std::string_view path, std::string_view username,
```

Comment:
Nit: You could do `pass = GetPassword()` and remote the above line

Reply:
Yes, I know I just thought this would be more verbose but IMO doesn't make a difference. I will address this